Hunger thirst systems

Slarvsylt opened the pull request:
System for thirst and hunger. When thirst/hunger reaches 1.0f a DeathEvent-component is attached to the animal. Also added a hybrid class for getting and decreasing hunger/thirst-levels.

Slarvsylt committed:
Began with thirst/hunger-system

Slarvsylt committed:
Commit local changes so I can continue later

Slarvsylt committed:
Finished Hunger/Thirst-systems and and added a hybrid class for getting thirst/hunger

gimme commented on Assets/Scripts/ECS/Animal/Needs/ThirstSystem.cs:
```diff
@@ -0,0 +1,43 @@
+
+            }).ScheduleParallel();
+
+            throw new NotImplementedException();
```
```suggestion
            m_EndSimulationEcbSystem.AddJobHandleForProducer(Dependency);
```

gimme commented on Assets/Scripts/ECS/Animal/Needs/HungerSystem.cs:
```diff
@@ -0,0 +1,44 @@
+                }
+
+            }).ScheduleParallel();
+            throw new NotImplementedException();
```
```suggestion

            m_EndSimulationEcbSystem.AddJobHandleForProducer(Dependency);
```

gimme commented on Assets/Scripts/ECS/Animal/Needs/HungerSystem.cs:
```diff
@@ -0,0 +1,44 @@
+            {
+                hungerData.Hunger += Time.DeltaTime / 1000.0f;
+
+                if(hungerData.Hunger <= 0.0f)
```
```suggestion
                if(hungerData.Hunger > 1.0f)
```

gimme commented on Assets/Scripts/ECS/Hybrid/NeedsStatus.cs:
```diff
@@ -0,0 +1,81 @@
+                entityManager.RemoveComponent<T>(entity);
+            }
+        }
+
```
```suggestion
```

gimme commented on Assets/Scripts/ECS/Hybrid/NeedsStatus.cs:
```diff
@@ -0,0 +1,81 @@
+        {
+            float cur = GetComp<ThirstData>().Thirst;
+            entityManager.SetComponentData<ThirstData>(entity, new ThirstData { Thirst = cur - value });
+        }
```
The decrease methods can reduce the values below 0. Is this intended?

gimme reviewed (commented):
I think that hunger/thirst would be easier to understand as like a "hunger bar" that **decreases** as you get **more** hungry.

I also think that it would be better if it was a number from 0 to infinity instead of 0 to 1 as that would work well with different max hungers for different animals.

Slarvsylt committed:
Update Assets/Scripts/ECS/Animal/Needs/ThirstSystem.cs

Co-Authored-By: Anders Magnusson <32415679+G1mmicK@users.noreply.github.com>

Slarvsylt committed:
Update Assets/Scripts/ECS/Animal/Needs/HungerSystem.cs

Co-Authored-By: Anders Magnusson <32415679+G1mmicK@users.noreply.github.com>

Slarvsylt committed:
Update Assets/Scripts/ECS/Hybrid/NeedsStatus.cs

Co-Authored-By: Anders Magnusson <32415679+G1mmicK@users.noreply.github.com>

Slarvsylt committed:
Some refactoring

Hunger and thirst now ranges from 0 to inifnity to better suit different animals max hinger/thirst. 0 == death

Slarvsylt commented:
I comply with your requests.
